### PR TITLE
Try moving blocks to their own modular directories

### PR DESCRIPTION
**Pull request by johnstonphilip**
## Tasks

- [x] I have signed a [Contributor License Agreement (CLA)](https://github.com/studiopress/pattern-manager/wiki/For-Developers#contributor-license-agreement-cla) with WP Engine.

### Summary of changes
This PR refactors the Pattern Block's (aka Patternception's) code into its own dedicated directory. 

### How to test
1. Just make sure the Pattern Blocks works as it should. No changes should actually happen with this PR.

### CODE REVIEW

File: wp-modules/editor/js/src/index.ts [hunk 1]
```diff
@@ -25,11 +25,6 @@ addFilter(
 	'pattern-manager/preventTransform',
 	preventTransform
 );
-addFilter(
-	'blocks.registerBlockType',
-	'pattern-manager/registerPatternBlock',
-	registerPatternBlock
-);
```
Comment (kienstra) [inline wp-modules/editor/js/src/index.ts:32]: Moving `addFilter()` out of this file is the only part of this PR I don't like. 

The idea of this file was to run all of the side-effects of the `editor/` module.

So importing another file shouldn't run side-effects.

Comment (kienstra) [reply wp-modules/editor/js/src/index.ts:32]: Still, it's just my opinion 😄

File: wp-modules/editor/js/src/blocks/pattern-block/registerPatternBlock.ts [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+import { addFilter } from '@wordpress/hooks';
+import PatternEdit from './PatternEdit';
+
+addFilter(
+	'blocks.registerBlockType',
+	'pattern-manager/registerPatternBlock',
+	registerPatternBlock
+);
 
 export default function registerPatternBlock(
```
Comment (kienstra) [inline wp-modules/editor/js/src/blocks/pattern-block/registerPatternBlock.ts:11]: Good idea to move the block-related functions to `blocks/`

File: wp-modules/editor/js/src/blocks/pattern-block/test/PatternEdit.tsx [hunk 3]
```diff
@@ -18,11 +18,11 @@ jest.mock( '../../globals', () => {
 	};
 } );
 
-jest.mock( '../../hooks/useSavedPostData', () => {
+jest.mock( '../../../hooks/useSavedPostData', () => {
 	return () => ( {} );
 } );
 
-jest.mock( '../../../../../app/js/src/components/PatternPreview', () => {
+jest.mock( '../../../../../../app/js/src/components/PatternPreview', () => {
```
Comment (kienstra) [inline wp-modules/editor/js/src/blocks/pattern-block/test/PatternEdit.tsx:25]: Mocking for React components is so messy 🤣

**Review by kienstra — APPROVED** [review]
Hi @johnstonphilip,
Good idea to move block functions to their own directory.

The Pattern Block works the same as before.

File: wp-modules/editor/js/src/index.ts (hunk 1 above)
Comment (johnstonphilip) [reply wp-modules/editor/js/src/index.ts:32]: Yeah I can see the argument for both ways. My preference is to be able to remove the whole block by simply deleting a directory. But I'd be alright with putting it back.

Comment (kienstra) [reply wp-modules/editor/js/src/index.ts:32]: Sounds good, it's a minor issue 😄 Whatever you'd like.

Comment (johnstonphilip) [reply wp-modules/editor/js/src/index.ts]: I moved it back :)

Comment (kienstra) [reply wp-modules/editor/js/src/index.ts]: Thanks a lot!

**Review by mike-day — APPROVED** [review]
This looks good, @johnstonphilip!

The structure is a bit easier to follow this way — it's nice cleanup.

**Pull request merged**